bin/fetch-configlet

```diff
@@ -76,18 +81,19 @@ def fetch_configlet():
         delta = wait_until - datetime.now(timezone.utc)
         seconds = delta.total_seconds()
         wait = seconds + 5 if seconds > BACKOFF_TIME else BACKOFF_TIME
-        print(f"Rate limited, sleeping {wait} seconds.")
+        logger.info(f"Rate limited, sleeping {wait} seconds.")
         time.sleep(wait)
         return 1
     if status >= 400:
-        sys.exit(f"Received unexpected {status} response from API.")
+        logger.error(f"Received unexpected {status} response from API: {response}")
+        return 1
```

Comment:
FWIW, I originally did that as `sys.exit()` because my thinking was a client error other than a 403 from the API is probably a permanent condition within the lifetime of the script run. Probably doesn't matter either way.

Reply:
Intent noted. It still seems the wrong approach to allow the function itself to trigger a full program exit. If we wanted to keep that short-circuit functionality, the best approach would probably be to raise some sort of unique exception to allow the caller to handle the failure however it wishes.